lib/active_merchant/billing/gateways/square.rb

```diff
@@ -0,0 +1,375 @@
+          add_address(post, options, :address)
+          r.process{ commit(:post, 'customers', post) }
+        end
+      end
+
+      # Scrubbing removes the access token from the header and the card_nonce.
+      # Square does not let the merchant ever see PCI data. All payment card data is directly
+      # handled on Square's servers via iframes as described here:
+      # https://docs.connect.squareup.com/articles/adding-payment-form/
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((Authorization: Bearer )[^\r\n]+), '\1[FILTERED]').
+          # Extra [\\]* for test. We do an extra escape in the regex of [\\]* b/c the remote_square_test.rb
+          # seems to double escape the backslashes before the quote. This ensures tests pass.
+          gsub(%r((\"card_nonce[\\]*\":[\\]*")[^"]+), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_address(post, options, non_billing_addr_key = :shipping_address)
+        if address = options[:billing_address] || options[:address]
+          add_address_for(post, address, :billing_address)
+        end        
+        non_billing_addr_key = non_billing_addr_key.to_sym
+        if address = options[non_billing_addr_key] || options[:address]
+          add_address_for(post, address, non_billing_addr_key)
+        end
+      end
+
+      def add_address_for(post, address, addr_key)
+          addr_key = addr_key.to_sym
+          post[addr_key] ||= {} # Or-Equals in case they passed in using Square's key format
+          post[addr_key][:address_line_1] = address[:address1] if address[:address1]
+          post[addr_key][:address_line_2] = address[:address2] if address[:address2]
+          post[addr_key][:address_line_3] = address[:address3] if address[:address3]
+          
+          post[addr_key][:locality] = address[:city] if address[:city]
+          post[addr_key][:sublocality] = address[:sublocality] if address[:sublocality]
+          post[addr_key][:sublocality_2] = address[:sublocality_2] if address[:sublocality_2]
+          post[addr_key][:sublocality_3] = address[:sublocality_3] if address[:sublocality_3]
+
+          post[addr_key][:administrative_district_level_1] = address[:state] if address[:state]
+          post[addr_key][:administrative_district_level_2] = address[:administrative_district_level_2] if address[:administrative_district_level_2] # In the US, this is the county.
+          post[addr_key][:administrative_district_level_3] = address[:administrative_district_level_3] if address[:administrative_district_level_3] # Used in JP not the US
+          post[addr_key][:postal_code] = address[:zip] if address[:zip]
+          post[addr_key][:country] = address[:country] if address[:country]
+      end
+
+      def add_amount(post, money, options)
+        post[:amount_money] = {}
+        post[:amount_money][:amount] = amount(money).to_i
```

Comment:
I would recommend `Integer(amount(money))` if you want to ensure strictness (e.g. `'2a'.to_i == 2`)